src/item_factory.cpp

```diff
@@ -1744,11 +1754,46 @@ void Item_factory::load_pet_armor( const JsonObject &jo, const std::string &src
 
 void islot_armor::load( const JsonObject &jo )
 {
-    optional( jo, was_loaded, "encumbrance", encumber, 0 );
-    // Default max_encumbrance will be set to a reasonable value in
-    // finalize_post
-    optional( jo, was_loaded, "max_encumbrance", max_encumber, -1 );
-    optional( jo, was_loaded, "coverage", coverage, 0 );
+    if( jo.has_array( "coverage_data" ) ) {
+
+        for( const JsonObject &obj : jo.get_array( "coverage_data" ) ) {
+            random_armor_data tempData;
+            body_part_set temp_cover_data;
+            assign_coverage_from_json( obj, "parts", temp_cover_data, sided );
+            tempData.covers = temp_cover_data;
+
+            if( obj.has_array( "encumbrance" ) ) {
+                tempData.encumber = obj.get_array( "encumbrance" ).get_int( 0 );
+                tempData.max_encumber = obj.get_array( "encumbrance" ).get_int( 1 );
+            } else if( obj.has_int( "encumbrance" ) ) {
+                tempData.encumber = obj.get_int( "encumbrance" );
+                tempData.max_encumber = obj.get_int( "encumbrance" );
+            }
+            tempData.coverage = obj.get_int( "coverage" );
+            data.push_back( tempData );
+
+            if( obj.has_int( "layer" ) ) {
+                for( auto &piece : data ) {
+                    piece.layer.value() = static_cast<layer_level>( obj.get_int( "layer" ) );
```

Comment:
```suggestion
                    piece.layer = static_cast<layer_level>( obj.get_int( "layer" ) );
```
Calling `value` for a `optional` is only allowed when actually contains a value. This is not guaranteed here. But you're not interested in the existing value (if any) anyway. Just set the new value on the `optional` directly like you do it in the loop below. (Why is this assigning the value of `pieces.layer` differently anyway?)